benches/smt.rs

```diff
@@ -0,0 +1,64 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use miden_crypto::{merkle::SimpleSmt, Felt};
+use rand_utils::prng_array;
+
+fn smt_rpo(c: &mut Criterion) {
+    // parameters
+
+    const DEPTH: u32 = 15;
+    const LEAVES: u32 = (1 << DEPTH) - 1;
+    const KEY: u64 = (LEAVES as u64) >> 2;
+
+    let mut seed = [0u8; 32];
+
+    // setup tree
+
+    let leaves: Vec<_> = (0..LEAVES)
+        .map(|_| {
+            let words: [u64; 4] = prng_array(seed);
+            seed = prng_array(seed);
+            [
+                Felt::new(words[0]),
+                Felt::new(words[1]),
+                Felt::new(words[2]),
+                Felt::new(words[3]),
+            ]
+        })
+        .collect();
+    let keys = leaves.iter().enumerate().map(|(i, _)| i as u64).collect();
+    let mut tree = SimpleSmt::new(keys, leaves, DEPTH).unwrap();
+
+    seed = prng_array(seed);
+    let words: [u64; 4] = prng_array(seed);
+    let leaf = [
+        Felt::new(words[0]),
+        Felt::new(words[1]),
+        Felt::new(words[2]),
+        Felt::new(words[3]),
+    ];
+
+    // benchmarks
+
+    let mut insert = c.benchmark_group(format!("smt update depth {DEPTH}"));
```

Comment:
minor nit: should we change the name to "smt update_leaf (depth{DEPTH})".

Reply:
Done!